src/render/painter.js

```diff
@@ -827,6 +875,18 @@ class Painter {
             timeStamps: this.loadTimeStamps
         };
     }
+
+    averageElevationNeedsEasing() {
+        if (!this.transform._elevation) return false;
+
+        const fog = this.style && this.style.fog;
+        if (!fog) return false;
+
+        const fogOpacity = fog.getPitchFactor(this.transform.pitch);
+        if (fogOpacity === 0) return false;
```

Comment:
I think we also need to check the fog color.

```suggestion
        if (fogOpacity === 0) return false;
        
        if  (fog.properties.get('color').a === 0) return false;
```

Reply:
Addressed in https://github.com/mapbox/mapbox-gl-js/pull/10564/commits/5b2538b30f14eeea1bac520447625111a68bc13b